dev-packages/test-utils/src/playwright-config.ts

```diff
@@ -0,0 +1,94 @@
+import type { PlaywrightTestConfig } from '@playwright/test';
+
+/** Get a playwright config to use in an E2E test app. */
+export function getPlaywrightConfig(
+  options: {
+    startCommand: string;
+    port?: number;
+    eventProxyPort?: number;
+    eventProxyFile?: string;
+  },
+  overwriteConfig?: Partial<PlaywrightTestConfig>,
+): PlaywrightTestConfig {
+  const testEnv = process.env['TEST_ENV'] || 'production';
+  const appPort = options.port || 3030;
+  const eventProxyPort = options.eventProxyPort || 3031;
+  const eventProxyFile = options.eventProxyFile || 'start-event-proxy.mjs';
+  const { startCommand } = options;
+
+  /**
+   * See https://playwright.dev/docs/test-configuration.
+   */
+  const config: PlaywrightTestConfig = {
+    testDir: './tests',
+    /* Maximum time one test can run for. */
+    timeout: 150_000,
+    expect: {
+      /**
+       * Maximum time expect() should wait for the condition to be met.
+       * For example in `await expect(locator).toHaveText();`
+       */
+      timeout: 10000,
+    },
+    fullyParallel: false,
+    workers: 1,
+    /* Fail the build on CI if you accidentally left test.only in the source code. */
+    forbidOnly: !!process.env.CI,
+    /* In dev mode some apps are flaky, so we allow retry there... */
+    retries: testEnv === 'development' ? 3 : 0,
+    /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+    reporter: 'list',
+    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+    use: {
+      /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
+      actionTimeout: 0,
+      /* Base URL to use in actions like `await page.goto('/')`. */
+      baseURL: `http://localhost:${appPort}`,
+
+      /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+      trace: 'on-first-retry',
+    },
+
+    /* Configure projects for major browsers */
+    projects: [
+      {
+        name: 'chromium',
+        use: {
+          // This comes from `devices["Desktop Chrome"]
+          userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.6422.26 Safari/537.36',
+          viewport: { width: 1280, height: 720 },
+          screen: { width: 1920, height: 1080 },
+          deviceScaleFactor: 1,
+          isMobile: false,
+          hasTouch: false,
+          defaultBrowserType: 'chromium',
+        },
+      },
+    ],
+
+    /* Run your local dev server before starting the tests */
+    webServer: [
+      {
+        command: `node ${eventProxyFile}`,
```

Comment:
there is probably some way we can make this event-proxy starting thing more convenient, but it's fine to do this as a followup.

Reply:
yeah, agreed, I guess if it is abstracted away this way it is less of a problem at least!